trunner/target/armv7m4.py

```diff
@@ -24,9 +25,26 @@
 
 
 class ARMv7M4Rebooter(Rebooter):
-    # TODO add text mode
-    # NOTE: changing boot modes not needed/supported for this target
-    pass
+    def _reboot_dut_text(self, hard):
+        # self.dut.send(" reboot\r\n")
+        subprocess.run(
+            [
+                "openocd",
+                "-f",
+                "interface/stlink.cfg",
+                "-f",
+                "target/stm32l4x.cfg",
+                "-c",
+                "reset_config srst_only srst_nogate connect_assert_srst",
+                "-c",
+                "init;reset;exit",
+            ],
+            encoding="ascii",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            timeout=20,
+            cwd=os.getcwd(),
```

Comment:
I wouldn't use the os module, since `TargetBase` should have specific paths to certain locations created (for example `self.project_dir`).

Also, I'm thinking if we should check if the reset was successful

Reply:
Sadly, this is not possible, OpenOCD throw non-zero status even if MCU voltage reading is below 3.3v (In our projects this reading is not even used and causing non-zero status return)